### PR TITLE
Puppet 5 Modules for API Manager 2.6.0 

### PR DESCRIPTION
**Pull request by irunika**

### CODE REVIEW

File: modules/apim/templates/carbon-home/repository/conf/carbon.xml.erb [hunk 1]
```diff
@@ -374,6 +401,23 @@
             <KeyPassword><%= @key_store['key_password'] %></KeyPassword>
         </KeyStore>
 
+        <!--
+            The KeyStore which is used for encrypting/decrypting internal data.
+            This block is read by Carbon Crypto Service.
+        -->
+        <InternalKeyStore>
```
Comment (VimukthiPerera) [inline modules/apim/templates/carbon-home/repository/conf/carbon.xml.erb:408]: @irunika I think Internal keystore should also be parameterized. WDYT?

File: modules/apim/templates/wso2am.service.erb [hunk 2]
```diff
@@ -1,5 +1,5 @@
 [Unit]
-Description=WSO2AM
+Description=WSO2AM - Dafult Profile
```
Comment (VimukthiPerera) [inline modules/apim/templates/wso2am.service.erb]: Spellings mistake. Should be "Default"

File: modules/apim_analytics_dashboard/templates/carbon-home/conf/dashboard/deployment.yaml.erb [hunk 3]
```diff
@@ -0,0 +1,383 @@
...
+        password: wso2carbon
+        driverClassName: org.h2.Driver
+        maxPoolSize: 10
+        idleTimeout: 60000
+        connectionTestQuery: SELECT 1
+        validationTimeout: 30000
+        isAutoCommit: false
+
+  #Data source for APIM Analytics
+  - name: APIM_ANALYTICS_DB
```
Comment (VimukthiPerera) [inline modules/apim_analytics_dashboard/templates/carbon-home/conf/dashboard/deployment.yaml.erb:321]: AFAIU, APIM_ANALYTICS_DB should be parameterized.

File: modules/apim_analytics_dashboard/templates/carbon-home/conf/dashboard/deployment.yaml.erb [hunk 4]
```diff
@@ -0,0 +1,383 @@
...
+    - name: "server.bootstrap.socket.timeout"
+      value: 60
+    - name: "client.bootstrap.socket.timeout"
+      value: 60
+    - name: "latency.metrics.enabled"
+      value: true
+
+  listenerConfigurations:
+    - id: "default-https"
+      host: "0.0.0.0"
```
Comment (VimukthiPerera) [inline modules/apim_analytics_dashboard/templates/carbon-home/conf/dashboard/deployment.yaml.erb]: hostname should also be parametrized

File: modules/apim_analytics_dashboard/templates/carbon-home/conf/dashboard/deployment.yaml.erb [hunk 5]
```diff
@@ -0,0 +1,383 @@
...
+
+  - name: WSO2_PERMISSIONS_DB
+    description: The datasource used for dashboard feature
+    jndiConfig:
+      name: jdbc/PERMISSION_DB
+      useJndiReference: true
+    definition:
+      type: RDBMS
+      configuration:
+        jdbcUrl: 'jdbc:h2:${sys:carbon.home}/wso2/${sys:wso2.runtime}/database/PERMISSION_DB;IFEXISTS=TRUE;DB_CLOSE_ON_EXIT=FALSE;LOCK_TIMEOUT=60000;MVCC=TRUE'
```
Comment (VimukthiPerera) [inline modules/apim_analytics_dashboard/templates/carbon-home/conf/dashboard/deployment.yaml.erb]: WSO2_PERMISSIONS_DB is parameterized in worker profile, but not here.

**Review by VimukthiPerera — APPROVED** [review]

**Review by DilanUA — APPROVED** [review]

**Pull request merged**